e2eTest/contract_deployments_test.go

```diff
@@ -141,6 +141,160 @@ func TestContractRedeployment(t *testing.T) {
 
 		require.Equal(t, []string{"HelloWorld"}, accountResp.Account.DeployedContracts)
 	})
+
+	t.Run("Contract redeployment with resource", func(t *testing.T) {
```

Comment:
this test doesn't really test what you would like to... I suggest you create a test where you have a contract that creates a resource, then make an update where that resource type is removed and then get the accounts so the storage inspection is called. Assert that it fails with dissabled rollback first.

The contract can be something like:

```
pub contract Person {
    pub fun makeFriends(): @Friendship {
        return <-create Friendship()
    }

    pub resource Friendship {
        pub fun yaay() {
            log("such a nice friend") // we can log to output, useful on emualtor for debugging
        }
    }
}
```

Deploy it, then execute a transaction calling the `makeFriends`:

```
import Person from 0x01 (or wherever it was deployed to)

transaction {
    let acc: AuthAccount

    prepare(signer: AuthAccount) {
        self.acc = signer    
    }
    
    execute {
        self.acc.save<@Person.Friendship>(<-Person.makeFriends(), to: StoragePath(identifier: "friendship")!)
    }
}

```

Then make an update to the contract like so:

```
pub contract Person { 
// empty
}
```

Then get the projects so account storage is inspected. This should fail without rollback, which you should confirm, and then add rollback and see if it works.

Reply:
There's no point where account inspection should fail in the test though since it's not optional to enable/ disable the rollback. This just happens on redeployment.

Reply:
I meant comment it out to validate test, and then uncomment to see the test works... a bit of turned around TDD

Reply:
Already verified that 👌